dango/dex/src/execute.rs

```diff
@@ -162,8 +163,25 @@ fn cancel_orders(ctx: MutableCtx, order_ids: BTreeSet<OrderId>) -> anyhow::Resul
     let mut events = Vec::new();
 
     for order_id in order_ids {
-        let (((base_denom, quote_denom), direction, price, _), order) =
-            ORDERS.idx.order_id.load(ctx.storage, order_id)?;
+        // First try to load from resting orders, then from incoming orders storage.
+        // If found in neither or both through an error
+        let (((base_denom, quote_denom), direction, price, _), order) = match (
+            ORDERS.idx.order_id.may_load(ctx.storage, order_id)?,
+            INCOMING_ORDERS.may_load(ctx.storage, order_id)?,
+        ) {
+            (Some(x), None) => {
+                ORDERS.remove(ctx.storage, x.0.clone())?;
+                x
+            },
+            (None, Some(x)) => {
+                INCOMING_ORDERS.remove(ctx.storage, order_id);
+                x
+            },
+            (None, None) => return Err(anyhow!("order with id {order_id} not found")),
+            (Some(_), Some(_)) => {
+                return Err(anyhow!("order with id {order_id} found in both maps"))
+            },
```

Comment:
Use the `bail!` macro instead of `return Err(anyhow!)`

Also can you add backticks around order id, for better readability:

```diff
- order with id {order_id} not found
+ order with id `{order_id}` not found
```